src/lib/eina/eina_thread_posix.c

```diff
@@ -0,0 +1,312 @@
+/* EINA - EFL data type library
+ * Copyright (C) 2012 Cedric Bail
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library;
+ * if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "eina_types.h"
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
+#include <stdlib.h>
+
+#include "eina_config.h"
+#include "eina_lock.h" /* it will include pthread.h with proper flags */
+#include "eina_thread.h"
+#include "eina_sched.h"
+#include "eina_cpu.h"
+
+/* undefs EINA_ARG_NONULL() so NULL checks are not compiled out! */
+#include "eina_safety_checks.h"
+
+#include "eina_debug_private.h"
+
+#include <pthread.h>
+#include <errno.h>
+#ifndef _WIN32
```

Comment:
Is it really needed?

Reply:
This is in case it runs in mingw

Reply:
Won't mingw always use native threads?

Reply:
Now yes, but if someone wants to change the build files to test POSIX in a MinGW environment, it will work.

Reply:
oh, nice, make sense